woocommerce/woocommerce
# Fixes #38757 - pass time values on CSV importer

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #38757 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run unit tests found in `plugins/woocommerce/tests/legacy/unit-tests/importer/product.php`
2. Alternatively
    - Use CSV importer to import sample CSV from `plugins/woocommerce/tests/legacy/unit-tests/importer/sample.csv`
    - Confirm `date_on_sale_from` and `date_on_sale_to` values for these SKUs `WOOLOGO` and `WOOALBUM` match expected output.

```
wp wc product list --sku=WOOLOGO,WOOALBUM --user=admin --fields=date_on_sale_from,date_on_sale_to
+---------------------+---------------------+
| date_on_sale_from   | date_on_sale_to     |
+---------------------+---------------------+
| 2017-01-01T00:00:00 | 2030-01-01T00:00:00 |
| 2023-07-08T00:00:00 | 2023-07-13T09:10:00 |
+---------------------+---------------------+
```



### Changelog entry




<details>

#### Significance




#### Type



-   [] Fix - Fixes an existing bug

#### Message

Fixes #38758 - pass time values on CSV importer.

#### Comment 

Updates product CSV importer to support Unix timestamps and any datetime values in formats supported by PHP `strtotime()`.

</details>
